azure-mgmt-monitor/src/main/java/com/microsoft/azure/management/monitor/implementation/ActivityLogsImpl.java

```diff
@@ -0,0 +1,274 @@
+        this.responsePropertySelector.add("resourceProviderName");
+        return this;
+    }
+
+    @Override
+    public FilterDefinitionStages.WithResponsePropertyDefinition withResourceId() {
+        this.responsePropertySelector.add("resourceId");
+        return this;
+    }
+
+    @Override
+    public FilterDefinitionStages.WithResponsePropertyDefinition withStatus() {
+        this.responsePropertySelector.add("status");
+        return this;
+    }
+
+    @Override
+    public FilterDefinitionStages.WithResponsePropertyDefinition withSubmissionTimestamp() {
+        this.responsePropertySelector.add("submissionTimestamp");
+        return this;
+    }
+
+    @Override
+    public FilterDefinitionStages.WithResponsePropertyDefinition withSubStatus() {
+        this.responsePropertySelector.add("subStatus");
+        return this;
+    }
+
+    @Override
+    public FilterDefinitionStages.WithResponsePropertyDefinition withSubscriptionId() {
+        this.responsePropertySelector.add("subscriptionId");
+        return this;
+    }
+
+    @Override
+    public FilterDefinitionStages.WithSelect apply() {
+        return this;
+    }
+
+    @Override
+    public FilterDefinitionStages.WithExecute filterByResourceGroup(String resourceGroupName) {
+        this.filterString = String.format(" and resourceGroupName eq '%s'", resourceGroupName);
+        return this;
+    }
+
+    @Override
+    public FilterDefinitionStages.WithExecute filterByResource(String resourceId) {
+        this.filterString = String.format(" and resourceUri eq '%s'", resourceId);
+        return this;
+    }
+
+    @Override
+    public FilterDefinitionStages.WithExecute filterByResourceProvider(String resourceProviderName) {
+        this.filterString = String.format(" and resourceProvider eq '%s'", resourceProviderName);
+        return this;
+    }
+
+    @Override
+    public FilterDefinitionStages.WithExecute filterByCorrelationId(String correlationId) {
+        this.filterString = String.format(" and correlationId eq '%s'", correlationId);
+        return this;
+    }
+
+    @Override
+    public PagedList<EventData> execute() {
+        return listEventData(getOdataFilterString() + this.filterString);
+    }
+
+    @Override
+    public Observable<EventData> executeAsync() {
+        return listEventDataAsync(getOdataFilterString() + this.filterString);
+    }
+    private String getOdataFilterString() {
+        return String.format("eventTimestamp ge '%s' and eventTimestamp le '%s'",
+                this.queryStartTime.withZone(DateTimeZone.UTC).toString(ISODateTimeFormat.dateTime()),
+                this.queryEndTime.withZone(DateTimeZone.UTC).toString(ISODateTimeFormat.dateTime()));
+    }
+
+    private PagedList<EventData> listEventData(String filter) {
+        ArrayList<String> sortedList = new ArrayList<String>(this.responsePropertySelector);
+        Collections.sort(sortedList);
```

Comment:
use type of `this.responsePropertySelector` as `TreeSet` so that we don't have to make an explicit sort here.